src/components/pages/__tests__/dashboard.test.tsx

```diff
@@ -1,10 +1,37 @@
-import { render, screen } from '@testing-library/react'
+import { configureStore } from '@reduxjs/toolkit'
+import { render, renderHook, screen, waitFor } from '@testing-library/react'
+import { Provider } from 'react-redux'
+
+import { useGetProductsByVendorQuery } from '@/lib/redux/features/product/api.ts'
+import { api } from '@/lib/redux/services/api.ts'
+import { withWrappers } from '@/lib/testing/utils.tsx'
 
 import DashboardPage from '../dashboard.tsx'
 
 describe('<DashboardPage />', () => {
+  it('should handle null response in providesTags', async () => {
+    const store = configureStore({
+      reducer: {
+        [api.reducerPath]: api.reducer,
+      },
+      middleware: getDefaultMiddleware =>
+        getDefaultMiddleware().concat(api.middleware),
+    })
+
+    const wrapper = ({ children }: { children: React.ReactNode }) => (
+      <Provider store={store}>{children}</Provider>
+    )
+
+    const { result } = renderHook(() =>
+      useGetProductsByVendorQuery({ id: '-1' }), { wrapper },
```

Comment:
Why did you put this in here? I think you are trying to test DashboardPage and putting it here might not be relevant

Reply:
Actually I'm having difficulties to create a test testing if the return json is null. Wasn't really sure on which file test i should add it so i just drop it here for the a temporaery time. @FreeJ1nG Maybe you can give your opinion regarding the test made for testing the function.